mne/channels/layout.py

```diff
@@ -882,13 +888,32 @@ def read_montage(kind, ch_names=None, path=None, scale=True):
         dtype = [('type', 'S8'), ('name', 'S8'),
                  ('x', 'f8'), ('y', 'f8'), ('z', 'f8')]
         data = np.loadtxt(fname, dtype=dtype)
-        pos_ = data[data['type'].astype(np.str) == 'eeg']
-        pos = np.vstack((pos_['x'], pos_['y'], pos_['z'])).T
-        ch_names_ = pos_['name'].astype(np.str)
+        pos = np.vstack((data['x'], data['y'], data['z'])).T
+        ch_names_ = data['name'].astype(np.str)
     else:
         raise ValueError('Currently the "%s" template is not supported.' %
                          kind)
     selection = np.arange(len(pos))
+
+    if transform is True:
+        pos = apply_trans(als_ras_trans_mm, pos)
```

Comment:
Well the question is how well guessing works I guess... If m and mm are the only possibilities I think guessing would be safe. Are there montages out there in cm? That would make it a lot more unreliable.